lib/helpers.bash

```diff
@@ -92,762 +96,741 @@ alias reload_completion="$(_make_reload_alias completion completion)"
-          rm "${BASH_IT}/$subdirectory/enabled/${plugin##*/}"
-        else
-          rm "${BASH_IT}/enabled/${plugin_global##*/}"
-        fi
-    fi
-
-    _bash-it-clean-component-cache "${file_type}"
-
-    printf '%s\n' "$file_entity disabled."
-}
-
-_enable-plugin ()
-{
-    _about 'enables bash_it plugin'
-    _param '1: plugin name'
-    _example '$ enable-plugin rvm'
-    _group 'lib'
-
-    _enable-thing "plugins" "plugin" $1 $BASH_IT_LOAD_PRIORITY_DEFAULT_PLUGIN
-}
-
-_enable-alias ()
-{
-    _about 'enables bash_it alias'
-    _param '1: alias name'
-    _example '$ enable-alias git'
-    _group 'lib'
-
-    _enable-thing "aliases" "alias" $1 $BASH_IT_LOAD_PRIORITY_DEFAULT_ALIAS
-}
-
-_enable-completion ()
-{
-    _about 'enables bash_it completion'
-    _param '1: completion name'
-    _example '$ enable-completion git'
-    _group 'lib'
-
-    _enable-thing "completion" "completion" $1 $BASH_IT_LOAD_PRIORITY_DEFAULT_COMPLETION
-}
-
-_enable-thing ()
-{
-    cite _about _param _example
-    _about 'enables a bash_it component'
-    _param '1: subdirectory'
-    _param '2: file_type'
-    _param '3: file_entity'
-    _param '4: load priority'
-    _example '$ _enable-thing "plugins" "plugin" "ssh" "150"'
-
-    subdirectory="$1"
-    file_type="$2"
-    file_entity="$3"
-    load_priority="$4"
-
-    if [ -z "$file_entity" ]; then
-        reference "enable-$file_type"
-        return
-    fi
-
-    if [ "$file_entity" = "all" ]; then
-        typeset f $file_type
-        for f in "${BASH_IT}/$subdirectory/available/"*.bash
-        do
-            to_enable=$(basename $f .$file_type.bash)
-            if [ "$file_type" = "alias" ]; then
-              to_enable=$(basename $f ".aliases.bash")
-            fi
-            _enable-thing $subdirectory $file_type $to_enable $load_priority
-        done
-    else
-        typeset to_enable=$(command ls "${BASH_IT}/$subdirectory/available/"$file_entity.*bash 2>/dev/null | head -1)
-        if [ -z "$to_enable" ]; then
-            printf '%s\n' "sorry, $file_entity does not appear to be an available $file_type."
-            return
-        fi
+		enabled_file="${enabled_file%."${file_type}"*.bash}"
+
+		enabled=
+		_bash-it-component-item-is-enabled "${file_type}" "${enabled_file}" && enabled='x'
+		printf "%-20s %-10s %s\n" "${enabled_file}" "[${enabled:- }]" "$(metafor "about-$file_type" < "$f")"
+	done
+	printf '\n%s\n' "to enable $preposition $file_type, do:"
+	printf '%s\n' "$ bash-it enable $file_type  <$file_type name> [$file_type name]... -or- $ bash-it enable $file_type all"
+	printf '\n%s\n' "to disable $preposition $file_type, do:"
+	printf '%s\n' "$ bash-it disable $file_type <$file_type name> [$file_type name]... -or- $ bash-it disable $file_type all"
+}
+
+function _on-disable-callback() {
+	_about 'Calls the disabled plugin destructor, if present'
+	_param '1: plugin name'
+	_example '$ _on-disable-callback gitstatus'
+	_group 'lib'
+
+	callback="$1_on_disable"
```

Comment:
The underscore is a `[:word:]` character, thus the positional parameter should be separated from the rest of the text with curly braces. This was already there of course, but it could be addressed in such a rework.

Reply:
Good catch!